test/value_to.cpp

```diff
@@ -114,12 +126,50 @@ class value_to_test
             std::invalid_argument);
     }
 
+    void
+    testContainerHelpers()
+    {
+        {
+            std::vector<int> v;
+            detail::try_reserve(
+                v, 10, detail::reserve_implementation<decltype(v)>());
+            BOOST_TEST(v.capacity() >= 10);
+            BOOST_STATIC_ASSERT(std::is_same<
+                decltype(detail::inserter(
+                    v, detail::inserter_implementation<decltype(v)>())),
+                decltype(std::back_inserter(v)) >::value);
+        }
+        {
+            std::array<int, 2> arr;
+            detail::try_reserve(
+                arr, 2, detail::reserve_implementation<decltype(arr)>());
+        }
+        {
+            int n;
+            detail::try_reserve(
```

Comment:
You're testing an implementation detail directly, which is not something the user would ever do. Better to test this indirectly

Reply:
This is specifically a test for these implementation details. Indirect testing would be much more complicated, and probably a bit brittle, because this helpers are simply for memory allocation optimization purposes.